doc/releases/changelog-0.23.1.md

```diff
@@ -7,8 +7,11 @@
 * Fixed a bug enabling PennyLane to work with the latest version of Autoray.
   [(#2548)](https://github.com/PennyLaneAI/pennylane/pull/2548)
 
+* Deprecating ExpvalCost by adding a UserWarning.
+  [(#2571)](https://github.com/PennyLaneAI/pennylane/pull/2571)
+
 <h3>Contributors</h3>
 
 This release contains contributions from (in alphabetical order):
 
-Josh Izaac
+Josh Izaac, Korbinian Kottmann
```

Comment:
Hi @Qottmann, this should be added to the `doc/releases/changelog-dev.md` file. :slightly_smiling_face:

 The `doc/releases/changelog-0.23.1.md` file was used for the `v0.23.1` bug fix release.